server/container/util-java8/src/test/java/org/apache/james/util/OptionalUtilsTest.java

```diff
@@ -119,4 +119,44 @@ public void orShouldReturnFirstValueWhenBothValues() {
                 Optional.of(2)))
             .contains(1);
     }
+
+    @Test
+    public void containsDifferentShouldReturnFalseWhenEmpty() throws Exception {
+        assertThat(OptionalUtils.containsDifferent(Optional.empty(), "any")).isFalse();
```

Comment:
What happens when requestValue is null? When storeValue is null?

Reply:
- we used requestValue as Optional to avoid null and it does not make sense when one more times checking for null
- I added unit test for storeValue null